internal/proxysql/core.go

```diff
@@ -1,150 +1,231 @@
-}
+		clientset, err := kubernetes.NewForConfig(config)
+		if err != nil {
+			slog.Error("error", slog.Any("err", err))
+		}
 
-// Define a custom type to implement the Sort interface.
-type ByPodIP []PodInfo
+		p.clientset = clientset
+	}
 
-func (a ByPodIP) Len() int           { return len(a) }
-func (a ByPodIP) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
-func (a ByPodIP) Less(i, j int) bool { return a[i].PodIP < a[j].PodIP }
+	// stop signal for the informer
+	stopper := make(chan struct{})
+	defer close(stopper)
 
-func (p *ProxySQL) Core() {
-	interval := p.settings.Core.Interval
+	app := p.settings.Core.PodSelector.App
+	namespace := p.settings.Core.PodSelector.Namespace
 
-	slog.Info("Core mode initialized, running loop", slog.Int("interval", interval))
+	labelSelector := labels.Set(map[string]string{
+		"app": app,
+	}).AsSelector()
+
+	factory := informers.NewSharedInformerFactoryWithOptions(
+		p.clientset,
+		1*time.Second,
+		informers.WithNamespace(namespace),
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.LabelSelector = labelSelector.String()
+		}),
+	)
 
-	for {
-		p.coreLoop()
+	podInformer := factory.Core().V1().Pods().Informer()
 
-		time.Sleep(time.Duration(interval) * time.Second)
+	defer runtime.HandleCrash()
+
+	go factory.Start(stopper)
+
+	if !cache.WaitForCacheSync(stopper, podInformer.HasSynced) {
+		runtime.HandleError(fmt.Errorf("Timed out waiting for caches to sync"))
+		return
 	}
-}
 
-func (p *ProxySQL) coreLoop() {
-	pods, err := GetCorePods(p.settings)
+	_, err := podInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    p.podAdded,
+		UpdateFunc: p.podUpdated,
+	})
 	if err != nil {
-		slog.Error("Failed to get pod info", slog.Any("error", err))
+		fmt.Println(err)
+	}
 
+	// block the main go routine from exiting
+	<-stopper
+}
+
+// This function is needed to do bootstrapping. At first I was using podUpdated to do adds, but we would never
+// get the first pod to come up. This function will only be useful on the first core pod to come up, the rest will
+// be handled via podUpdated.
+//
+// This feels a bit clumsy.
+func (p *ProxySQL) podAdded(object interface{}) {
+	pod, ok := object.(*v1.Pod)
+	if !ok {
 		return
 	}
 
-	if len(pods) == 0 {
-		slog.Error("No pods returned")
-
+	// if the new pod is not THIS pod, bail out of this function. the rest of this function should only apply
+	// to the first core pod to come up in the cluster.
+	if hostname, _ := os.Hostname(); pod.Name != hostname {
 		return
 	}
 
-	checksumFile := "/tmp/pods-cs.txt"
-	digest := calculateChecksum(pods)
+	// check if pod is already in the proxysql_servers table; this can happen when core pods add
+	// other core pods.
+	query := fmt.Sprintf("SELECT count(*) FROM proxysql_servers WHERE hostname = '%s'", pod.Status.PodIP)
+
+	var count int
 
-	// Read the previous checksum from the file
-	old, err := os.ReadFile(checksumFile)
+	err := p.conn.QueryRow(query).Scan(&count)
```

Comment:
Should be able to do this:

```suggestion
	err := p.conn.QueryRow("SELECT count(*) FROM proxysql_servers WHERE hostname = $1", pod.Status.PodIP).Scan(&count)
```